runtime/drivers/bigquery/olap.go

```diff
@@ -0,0 +1,536 @@
+package bigquery
+
+import (
+	"context"
+	sqldriver "database/sql/driver"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math/big"
+	"reflect"
+	"time"
+
+	"cloud.google.com/go/bigquery"
+	"cloud.google.com/go/civil"
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+	"github.com/rilldata/rill/runtime/drivers"
+	"github.com/rilldata/rill/runtime/pkg/observability"
+	"go.uber.org/zap"
+	"google.golang.org/api/iterator"
+)
+
+var _ drivers.OLAPStore = (*Connection)(nil)
+
+// Dialect implements drivers.OLAPStore.
+func (c *Connection) Dialect() drivers.Dialect {
+	return drivers.DialectBigQuery
+}
+
+// Exec implements drivers.OLAPStore.
+func (c *Connection) Exec(ctx context.Context, stmt *drivers.Statement) error {
+	if c.config.LogQueries {
+		c.logger.Info("bigquery query", zap.String("sql", c.Dialect().SanitizeQueryForLogging(stmt.Query)), zap.Any("args", stmt.Args), observability.ZapCtx(ctx))
+	}
+	client, err := c.acquireClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	q := client.Query(stmt.Query)
+	q.Parameters = make([]bigquery.QueryParameter, len(stmt.Args))
+	for i, arg := range stmt.Args {
+		q.Parameters[i] = bigquery.QueryParameter{
+			Value: arg,
+		}
+	}
+	if stmt.DryRun {
+		q.DryRun = true
+	}
+
+	j, err := q.Run(ctx)
+	if err != nil {
+		return err
+	}
+	if stmt.DryRun {
+		// Dry run is not asynchronous
+		status := j.LastStatus()
+		return status.Err()
+	}
+	js, err := j.Wait(ctx)
+	if err != nil {
+		return err
+	}
+	err = js.Err()
+	return err
+}
+
+// InformationSchema implements drivers.OLAPStore.
+func (c *Connection) InformationSchema() drivers.OLAPInformationSchema {
+	return c
+}
+
+// MayBeScaledToZero implements drivers.OLAPStore.
+func (c *Connection) MayBeScaledToZero(ctx context.Context) bool {
+	return false
+}
+
+// Query implements drivers.OLAPStore.
+func (c *Connection) Query(ctx context.Context, stmt *drivers.Statement) (res *drivers.Result, resErr error) {
+	if c.config.LogQueries {
+		c.logger.Info("bigquery query", zap.String("sql", c.Dialect().SanitizeQueryForLogging(stmt.Query)), zap.Any("args", stmt.Args), observability.ZapCtx(ctx))
+	}
+	client, err := c.acquireClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	q := client.Query(stmt.Query)
```

Comment:
Since BigQuery has good dry run support, would this work here (as in the current `Exec`)?
```
	if stmt.DryRun {
		q.DryRun = true
	}
```